components/InputTypeLocation.js

```diff
@@ -151,16 +151,29 @@ class InputTypeLocation extends React.Component {
       return this.props.onChange(location);
     }
 
-    const countryComponent = value.gmaps['address_components'].find(c => c.types.includes('country'));
+    const country = value.gmaps['address_components'].find(c => c.types.includes('country'))?.['short_name'];
+
+    /* Use ADR microformat field `adr_address` because of more consistent formatting and since
+       it also includes a single field for street address (with house number in the correct place depending on locality) */
+    const adrAddress = value.gmaps['adr_address'];
+    const streetAddress = adrAddress.match(/<span class="street-address">([^<]+)<\/span>/i)?.[1];
+    const postalCode = adrAddress.match(/<span class="postal-code">([^<]+)<\/span>/i)?.[1];
+    const city = adrAddress.match(/<span class="locality">([^<]+)<\/span>/i)?.[1];
+    const zone = adrAddress.match(/<span class="region">([^<]+)<\/span>/i)?.[1];
```

Comment:
The [other option available from Google seems to be the `address_components`](https://developers.google.com/maps/documentation/places/web-service/details#Place-address_components) JSON which separates out house number from street so we'd have to choose where to put the house number (and uses different names for city/locality in different locales). 

Switched to using a DOMParser instead of regexp, nice enhancement, thanks!